FinalEngine.Tests/Editor/ViewModels/Docking/DockViewModelTests.cs

```diff
@@ -0,0 +1,398 @@
+                null,
+                this.sceneHierarchyFactory.Object,
+                this.propertiesFactory.Object,
+                this.consoleFactory.Object,
+                this.entitySystemsFactory.Object,
+                this.sceneViewFactory.Object);
+        });
+    }
+
+    [Test]
+    public void ConstructorShouldThrowArgumentNullExceptionWhenPropertiesFactoryIsNull()
+    {
+        // Act and assert
+        Assert.Throws<ArgumentNullException>(() =>
+        {
+            new DockViewModel(
+                this.logger.Object,
+                this.layoutManagerFactory.Object,
+                this.projectExplorerFactory.Object,
+                this.sceneHierarchyFactory.Object,
+                null,
+                this.consoleFactory.Object,
+                this.entitySystemsFactory.Object,
+                this.sceneViewFactory.Object);
+        });
+    }
+
+    [Test]
+    public void ConstructorShouldThrowArgumentNullExceptionWhenSceneHierarchyFactoryIsNull()
+    {
+        // Act and assert
+        Assert.Throws<ArgumentNullException>(() =>
+        {
+            new DockViewModel(
+                this.logger.Object,
+                this.layoutManagerFactory.Object,
+                this.projectExplorerFactory.Object,
+                null,
+                this.propertiesFactory.Object,
+                this.consoleFactory.Object,
+                this.entitySystemsFactory.Object,
+                this.sceneViewFactory.Object);
+        });
+    }
+
+    [Test]
+    public void ConstructorShouldThrowArgumentNullExceptionWhenSceneViewFactoryIsNull()
+    {
+        // Act and assert
+        Assert.Throws<ArgumentNullException>(() =>
+        {
+            new DockViewModel(
+                this.logger.Object,
+                this.layoutManagerFactory.Object,
+                this.projectExplorerFactory.Object,
+                this.sceneHierarchyFactory.Object,
+                this.propertiesFactory.Object,
+                this.consoleFactory.Object,
+                this.entitySystemsFactory.Object,
+                null);
+        });
+    }
+
+    [Test]
+    public void LoadCommandExecuteShouldInvokeLayoutManagerContainsLayoutWhenInvoked()
+    {
+        // Act
+        this.viewModel.LoadCommand.Execute(null);
+
+        // Assert
+        this.layoutManager.Verify(x => x.ContainsLayout("startup"), Times.Once);
+    }
+
+    [Test]
+    public void LoadCommandExecuteShouldInvokeLayoutManagerFactoryCreateManagerWhenInvoked()
+    {
+        // Act
+        this.viewModel.LoadCommand.Execute(null);
+
+        // Assert
+        this.layoutManagerFactory.Verify(x => x.CreateManager(), Times.Once);
+    }
+
+    [Test]
+    public void LoadCommandExecuteShouldInvokeLayoutManagerResetLayoutWhenDoesNotContainStartupLayout()
+    {
+        // Act
+        this.viewModel.LoadCommand.Execute(null);
+
+        // Assert
+        this.layoutManager.Verify(x => x.ResetLayout(), Times.Once);
```

Comment:
💡TIP💡

I personally got sick and tired of writing the `, Times.Once);` at the end of my mock verifies.  I wrote an extension method named `VerifyOnce()` so I did not have to anymore.  Of course this is only worth it if you decide to stay with Moq.

If you plan on moving to NSubstitute, don't bother.

Reply:
I plan on moving over to NSubstitute #238. Just have to work out exactly how I'm going to approach this and chip away at each test class most likely.